src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -125,6 +125,266 @@ private static void VerifyMissingType(string source, WellKnownType type, params
         private CSharpCompilation CreateCompilationWithAsyncIterator(CSharpTestSource source, CSharpCompilationOptions options = null)
             => CreateCompilationWithTasksExtensions(new[] { source, AsyncStreamsTypes }, options: options);
 
+        [Fact]
+        [WorkItem(38961, "https://github.com/dotnet/roslyn/issues/38961")]
+        public void LockInsideFinally()
+        {
+            var comp = CreateCompilationWithAsyncIterator(@"
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class C
+{
+    object _splitsLock = new object();
+    public async IAsyncEnumerable<string> GetSplits()
+    {
+        try
+        {
+        }
+        finally
+        {
+            lock (_splitsLock)
+            {
+                Console.WriteLine(""hello"");
+            }
+            Console.WriteLine(""world"");
+        }
+        yield break;
+    }
+}");
+
+            var v = CompileAndVerify(comp);
```

Comment:
Consider adding an `expectedOutput` argument for this test as well as `FinallyInsideFinally`